python/clx/eda/eda.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+
+import cuxfilter
+from cuxfilter.layouts import feature_and_double_base
+
+from clx.eda.summary_stats import SummaryStatistics
+
+
+class EDA:
+    eda_modules = {"SummaryStatistics": SummaryStatistics}
+
+    def __init__(self, dataframe):
+        """
+        An EDA (Exploratory Data Analysis) Object.
+        Parameters
+        ----------
+        :param dataframe: Dataframe to be used for analysis
+        :type dataframe: cudf.DataFrame
+
+        Examples
+        --------
+        >>> from clx.eda import EDA
+        >>> import cudf
+        >>> import pandas as pd
+        >>> df = cudf.DataFrame()
+        >>> df['a'] = [1,2,3,4]
+        >>> df['b'] = ['a','b','c','c']
+        >>> df['c'] = [True, False, True, True]
+        >>> df['d'] = cudf.Series(pd.date_range("2000-01-01", periods=3,freq="m"))
+        >>> eda = EDA(df)
+        >>> eda
+            {
+                "SummaryStatistics": {
+                    "a": {
+                        "dtype": "int64",
+                        "summary": {
+                            "unique": "4",
+                            "total": "4"
+                        }
+                    },
+                    "b": {
+                        "dtype": "object",
+                        "summary": {
+                            "unique": "3",
+                            "total": "4"
+                        }
+                    },
+                    "c": {
+                        "dtype": "bool",
+                        "summary": {
+                            "true_percent": "0.75"
+                        }
+                    },
+                    "d": {
+                        "dtype": "datetime64[ns]",
+                        "summary": {
+                            "timespan": "60 days, 2880 hours, 0 minutes, 0 seconds"
+                        }
+                    }
+                }
+            }
+        """
+        self.__dataframe = dataframe
+        self.__module_ref = {}
```

Comment:
Can we remove the initialization of `self. __module_ref` and `self. __analysis` in the constructor as initialization happening in the `__generate_analysis` function?